### PR TITLE
Fix Audit Lock mechanism

### PR DESCRIPTION
**Pull request by adecaro**
This PR fixes the audit lock mechanism.
Before, the unlock could be delayed by up 4 minutes if the sender does not send the envelope due to
some failure happening in the meantime.
Now, the locks are released as soon as the auditor approves the transaction.

Signed-off-by: Angelo De Caro <adc@zurich.ibm.com>

### CODE REVIEW

File: integration/token/interop/tests.go [hunk 1]
```diff
@@ -105,7 +110,18 @@ func TestHTLCSingleNetwork(network *integration.Infrastructure) {
 
 	CheckPublicParams(network, defaultTMSID, "issuer", "auditor", "alice", "bob")
 	CheckOwnerDB(network, defaultTMSID, nil, "issuer", "auditor", "alice", "bob")
-	CheckAuditorDB(network, token.TMSID{}, "auditor", "", nil)
+	CheckAuditorDB(network, token.TMSID{}, "auditor", "", func(errs []string) error {
+		fmt.Printf("Got errors [%v]", errs)
+		if len(errs) != 4 {
+			return errors.Errorf("expected 8 errors, got [%d]", len(errs))
```
Comment (KElkhiyaoui) [inline integration/token/interop/tests.go:116]: expected 4 errors

File: integration/token/interop/tests.go [hunk 2]
```diff
@@ -116,19 +132,25 @@ func TestHTLCSingleNetwork(network *integration.Infrastructure) {
...
 	)
+	htlcLock(network, defaultTMSID, "alice", "", "USD", 1, "bob", 1*time.Hour, nil, crypto.SHA3_256)
 
 	CheckPublicParams(network, defaultTMSID, "issuer", "auditor", "alice", "bob")
 	CheckOwnerDB(network, defaultTMSID, nil, "issuer", "auditor", "alice", "bob")
 	CheckAuditorDB(network, token.TMSID{}, "auditor", "", func(errs []string) error {
 		fmt.Printf("Got errors [%v]", errs)
-		if len(errs) != 8 {
+		if len(errs) != 12 {
 			return errors.Errorf("expected 8 errors, got [%d]", len(errs))
```
Comment (KElkhiyaoui) [inline integration/token/interop/tests.go:142]: expected 12 errors

Comment (adecaro) [reply integration/token/interop/tests.go:142]: done

File: integration/token/interop/tests.go (hunk 1 above)
Comment (adecaro) [reply integration/token/interop/tests.go:116]: done

**Review by KElkhiyaoui — APPROVED** [review]

**Pull request merged**